src/TTextEdit.cpp

```diff
@@ -1151,9 +1188,13 @@ void TTextEdit::mousePressEvent( QMouseEvent * event )
     }
     if( event->button() == Qt::LeftButton )
     {
+        if( QGuiApplication::queryKeyboardModifiers().testFlag(Qt::ControlModifier) ) {
```

Comment:
If I recall correctly the keyboard modifiers are available as `QMouseEvent::modifiers()` so you can use the more concise:
`if( event->modifiers() & Qt::ControlModifier ) {`